route_following_plugin/src/route_following_plugin.cpp

```diff
@@ -138,44 +136,48 @@ namespace route_following_plugin
             double end_dist = wm_->routeTrackPos(route_shortest_path.back().centerline2d().back()).downtrack; 
             maneuvers.push_back(composeLaneFollowingManeuverMessage(start_dist, end_dist, prev_mvr_end_speed, target_speed_in_lanelet, route_shortest_path.back().id(),ros::Time::now()));
         }
-        
+        ROS_DEBUG_STREAM("Maneuver plan along route successfully generated");
         return maneuvers;
     }
 
     bool RouteFollowingPlugin::plan_maneuver_cb(cav_srvs::PlanManeuversRequest &req, cav_srvs::PlanManeuversResponse &resp)
     { 
         if(latest_maneuver_plan_.empty()){
             ROS_ERROR_STREAM("A maneuver plan has not been generated");
-            return true;
+            return false;
         }
-
-        lanelet::BasicPoint2d current_loc(pose_msg_.pose.position.x, pose_msg_.pose.position.y);
         
-        double current_downtrack = wm_->routeTrackPos(current_loc).downtrack;
+        double current_downtrack = wm_->routeTrackPos(current_loc_).downtrack;
         double speed_progress = current_speed_;
         
         //update start distance of first maneuver and time for all maneuvers
         setManeuverStartDist(latest_maneuver_plan_.front(), current_downtrack);
         //Update time progress for maneuvers
         updateTimeProgress(latest_maneuver_plan_, ros::Time::now());
-        //update starting speed
         
         //Return the set of maneuvers which intersect those 15s
         int i=0;
         double planned_time = 0.0;
         while(planned_time < min_mvr_duration_ && i < latest_maneuver_plan_.size()){
+            //Ignore plans for distance already covered
+            if(GET_MANEUVER_PROPERTY(latest_maneuver_plan_[i], end_dist) < current_downtrack){
+                ++i;
+                continue;
+            }
             double maneuver_start_time = GET_MANEUVER_PROPERTY(latest_maneuver_plan_[i], start_time).toSec();
             double maneuver_end_time = GET_MANEUVER_PROPERTY(latest_maneuver_plan_[i],end_time).toSec();
             planned_time += maneuver_end_time - maneuver_start_time;
```

Comment:
I believe this computation of planned_time is erroneous if we are already in the middle of this lanelet/maneuver.  It should only add the time remaining to complete the lanelet if time_progress > maneuver_start_time.